src/main/java/knn/api/APIKnowledgeNodeOut.java

```diff
@@ -0,0 +1,10 @@
+package knn.api;
+
+import prometheus.api.GeneralOutputAPI;
+
+/**
+ * Meant to handle output of API.
+ */
+public abstract class APIKnowledgeNodeOut implements GeneralOutputAPI {
```

Comment:
```suggestion
public abstract class APIKnowledgeNodeOutput implements GeneralOutputAPI {
```
Little bit of nitpicking there lol

Edit: Following what I wrote in the APIKnowledgeNoee class, this class might not be required. Depends from what the prof wants